### ISSUE TITLE
How to Send Headers to Web API using Canopy?

### ISSUE THREAD

**Issue by kunalthecoder** [lefthandedgoat/canopy#271]
Hi,

I would like to send some headers to Web API which I am calling from one Web API.

Kunal

**Comment by amirrajan** [comment]
Use [Http.fs](https://github.com/relentless/Http.fs)?

**Comment by lefthandedgoat** [comment]
Http.fs will work for you like amir suggested.

**Comment by kunalthecoder** [comment]
Hi All

Thanks for your Reply!

I am trying to make request through canopy. Which will browse URL of Web API and In that I would like to pass Headers.

How to create complete request object along with headers and assign that to browser which we will open from Canopy.

Something like 

url request; // request will contains headers also.

Thanks,
Kunal

**Comment by lefthandedgoat** [comment]
I dont think that you can.  Read more here: http://stackoverflow.com/questions/15645093/setting-request-headers-in-selenium

**Comment by kunalthecoder** [comment]
Let me try it out!

Thanks

**Comment by kunalthecoder** [comment]
I tried it out but it could not help much!

I tried something like below.

Step 1: Open Browser with your API URL

Step 2: Delete the response received from that API call in browser and make browser screen blank.

Note: URL using which you are opening the browser it will return some error as response, as we are not passing the required headers so it may result into missing specific header error.

Step 3: Make Java script call through HttpClient and Set body of Browser with response received from HttpClient Call.

Step 4: Validate the  response received in browser using Canopy "Contains".

Thanks,
Kunal
